sklearn/neighbors/tests/test_lof.py

```diff
@@ -0,0 +1,63 @@
+from math import sqrt
+import numpy as np
+from sklearn import neighbors
+
+from numpy.testing import assert_array_equal
+
+from sklearn.metrics import roc_auc_score
+from sklearn.utils import check_random_state
+from sklearn.utils.testing import assert_greater
+from sklearn.utils.testing import assert_array_almost_equal
+
+
+def test_lof():
+    # Toy sample (the last two samples are outliers):
+    X = [[-2, -1], [-1, -1], [-1, -2], [1, 1], [1, 2], [2, 1], [5, 3], [-4, 2]]
+
+    # Test LocalOutlierFactor:
+    clf = neighbors.LocalOutlierFactor()
+    score = -clf.fit(X).decision_function()
+    assert_array_equal(clf._fit_X, X)
+
+    # Assert scores are good:
+    assert_greater(np.min(score[-2:]), np.max(score[:-2]))
+
+    # Assert predict() works:
+    clf = neighbors.LocalOutlierFactor(contamination=0.25).fit(X)
+    assert_array_equal(clf.predict(), 6 * [1] + 2 * [-1])
```

Comment:
even given the discussion above? (your comment on Feb 23th: "`ok with X= None`" !!!)